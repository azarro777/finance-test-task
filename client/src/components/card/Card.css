.card-container {
  width: 175px;
  height: 46px;
  background-color: #ffffff;
  align-items: center;
  border-radius: 8px;
  border: 1px solid #dadce0;
  white-space: nowrap;
  display: flex;
  margin: 2rem 0;
}

.card-container:hover {
  cursor: pointer;
  background-color: #f8f9fa;
  border: 1px solid #ced0d4;
}

.card-title {
  display: flex;
  flex-direction: column;
  width: 80px;
  margin-left: 1rem;
  align-items: flex-start;
}

.card-title__ticker {
  margin: 0;
  padding: 0;
  color: #202124;
  font-weight: bold;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 0.75rem;
  line-height: 1rem;
}

.card-title__price {
  margin: 0;
  padding: 0;
  color: #3c4043;
  font-size: 0.75rem;
  font-weight: 400;
  line-height: 1rem;
}

.card-persent {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-right: 0.5rem;
}

.card-persent__persent,
.card-persent__change {
  margin: 0;
  padding: 0;
  color: #3c4043;
  font-size: 0.75rem;
  font-weight: 500;
  line-height: 1rem;
}