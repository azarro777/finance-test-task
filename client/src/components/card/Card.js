import { IconContext } from 'react-icons';
import { CgArrowUpR, CgArrowDownR } from 'react-icons/cg';
import './Card.css';

export const Card = (props) => {
	const {color, ticker, price, persent, change} = props;

  return (
    <div className="card-container">
      <IconContext.Provider
        value={{
          size: '2rem',
          color: color,
          className: 'global-class-name'
        }}
      >
        {color === '#137333' ? <CgArrowUpR /> : <CgArrowDownR />}
      </IconContext.Provider>
      <div className="card-title">
      <h3 className="card-title__ticker">{ticker}</h3>
      <p className="card-title__price" style={{color: color}}>
        {price}{' '}
      </p>
    </div>
      <div className="card-persent">
      <p className="card-persent__persent" style={{color: color}}>
        {persent}%
      </p>
      <p className="card-persent__change" style={{color: color}}>
        {change}
      </p>
    </div>
    </div>
  );
};