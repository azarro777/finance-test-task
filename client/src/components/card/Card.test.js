import React from 'react';
import { Card } from './Card';
import { shallow } from 'enzyme';

describe('Testing <Card/>', () => {
	it('Card have rendered correctly', () => {
		const card = shallow(<Card/>);
		expect(card).toMatchSnapshot();
	});
});