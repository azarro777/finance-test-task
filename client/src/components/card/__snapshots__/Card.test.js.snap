// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <Card/> Card have rendered correctly 1`] = `ShallowWrapper {}`;
